### ISSUE TITLE
Normalise (quote) strings when editing through the Yaml interface?

### ISSUE THREAD

**Issue by lockywolf** [lucc/khard#276]
I am opening a vcard that has passed the following route: khard(new)-vdirsyncer->Google-vdirsyncer->khard(edit).

```
BEGIN:VCARD
VERSION:3.0
N:KhardLastName;KhardTestName;;;
FN:KhardTestName KhardLastName
TITLE:PhD
ORG:KharderProject;Unit2
REV:2020-10-14T01:40:56Z
UID:7bb2d3d99283d675
TEL;TYPE=CELL:+0 000 0000000
TEL;TYPE=HOME:+0 000 1000000
NOTE:Khard note line 1. Khard note line 2.
item1.IMPP;X-SERVICE-TYPE=Skype:skype:from-phone-skype
X-DOMAIN-NAME:@khard-rss
X-FACEBOOK:@khard-facebook
X-JABBER:jabber@khard.er
X-MASTODON:@khard-mastodon
X-SKYPE:skype-kharder
X-TELEGRAM:@khard-telegram
X-TWITTER:@khard-twitter
item1.X-ABLabel:Other
END:VCARD
```

Too bad I didn't save the original .vcf created by khard, but the following is the issue:

I cannot edit this vcard with Yaml, because of the "@uid" values, as apparently Yaml tokens cannot begin with a "@" symbol. May I suggest enquoting values fed from vcf to yaml, unless already enquoted?
The other way would be ambiguous, but I think that removing the quotes is a better option.

**Comment by lucc** [comment]
Currently conversion to yaml is implemented by hand in the methods `khard.carddav_object.YAMLEditable.to_yaml` and `khard.helpers.convert_to_yaml`.  We could fix it there but the clean solution would be to use a library for generating yaml. (We are using `ruamel.yaml` to *parse* yaml.)   If somebody wants to propose a library or even a PR that would be interesting.

I think the requirements for generating yaml with a library are that we can put comments into the output.  We currently have that for documentation. There are proposals (#146) for less verbose yaml output but that should just be an option for advanced users. For newcomers it should be as well documented as it is now.  The special indent rules and special case formatting might not be so important to keep.

**Comment by skowalak** [comment]
I just tested reading and then writing the template with `ruamel.yaml` and the following script. It preserved the comments. Unfortunately, I am not very familiar with `khard`s codebase yet, but I could try creating a PR to use `ruamel.yaml` and quote yaml values.

``` python
import sys
from ruamel import yaml

with open("khard/data/template.yaml", "r") as template:
    yaml_template = template.read()
    ry = yaml.YAML()
    template_code = ry.load(yaml_template)
    ry.dump(template_code, sys.stdout)

```

Result:
``` yaml
# Every contact must contain a formatted name, it will be autofilled
# from the full name below if not given.
Formatted name:

# name components
# every entry may contain a string or a list of strings
# format:
#   First name : name1
#   Additional :
#       - name2
#       - name3
#   Last name  : name4
Prefix:
First name:
Additional:
Last name:
Suffix:

# nickname
# may contain a string or a list of strings
Nickname:

# important dates
# Formats:
#   vcard 3.0 and 4.0: yyyy-mm-dd or yyyy-mm-ddTHH:MM:SS
#   vcard 4.0 only: --mm-dd or text= string value
# anniversary
Anniversary:
# birthday
Birthday:

# organisation
# format:
#   Organisation : company
# or
#   Organisation :
#       - company1
#       - company2
# or
#   Organisation :
#       -
#           - company
#           - unit
Organisation:

# organisation title and role
# every entry may contain a string or a list of strings
#
# title at organisation
# example usage: research scientist
Title:
# role at organisation
# example usage: project leader
Role:

# phone numbers
# format:
#   Phone:
#       type1, type2: number
#       type3:
#           - number1
#           - number2
#       custom: number
# allowed types:
#   vcard 3.0: At least one of bbs, car, cell, fax, home, isdn, msg, modem,
#                              pager, pcs, pref, video, voice, work
#   vcard 4.0: At least one of home, work, pref, text, voice, fax, cell, video,
#                              pager, textphone
#   Alternatively you may use a single custom label (only letters).
#   But beware, that not all address book clients will support custom labels.
Phone:
  cell:
  home:

# email addresses
# format like phone numbers above
# allowed types:
#   vcard 3.0: At least one of home, internet, pref, work, x400
#   vcard 4.0: At least one of home, internet, pref, work
#   Alternatively you may use a single custom label (only letters).
Email:
  home:
  work:

# post addresses
# allowed types:
#   vcard 3.0: At least one of dom, intl, home, parcel, postal, pref, work
#   vcard 4.0: At least one of home, pref, work
#   Alternatively you may use a single custom label (only letters).
Address:
  home:
    Box:
    Extended:
    Street:
    Code:
    City:
    Region:
    Country:

# categories or tags
# format:
#   Categories : single category
# or
#   Categories :
#       - category1
#       - category2
Categories:

# web pages
# may contain a string or a list of strings
Webpage:

# private objects
# define your own private objects in the vcard section of your khard config file
# example:
#   [vcard]
#   private_objects = Jabber, Skype, Twitter
# these objects are stored with a leading "X-" before the object name in the
# vcard files.
# every entry may contain a string or a list of strings
Private: {}

# notes
# may contain a string or a list of strings
# for multi-line notes use:
#   Note : |
#       line one
#       line two
Note:

```

**Comment by lucc** [comment]
@skowalak I would welcome a PR :)

I have mentioned the code that is most relevant to this in the comment above.  But just ask questions when you need help understanding stuff or planing the design.

**Comment by skowalak** [comment]
@lucc I have added a PR draft, more details are inside #302

**Comment by lucc** [comment]
@lockywolf can you confirm that this is fixed by #302?
